cogs/monitor.py

```diff
@@ -214,29 +241,171 @@ async def user_delete(self, ctx: commands.Context, member: discord.Member):
     
     @monitor.command()
     @admin_command()
-    async def toggle(self, ctx: commands.Context):
+    async def toggle(self, ctx: commands.Context, type: Optional[Literal['messages','users']] = None):
         """
         Toggle if monitoring is active for the guild.
         """ 
         guild_data = (await GuildData.get_or_create(guild_id=ctx.guild.id))[0]
-        guild_data.monitoring = not guild_data.monitoring
+
+        if not type or type == "messages":
+            guild_data.monitoring_message = not guild_data.monitoring_message
+
+        if not type or type == "users":
+            guild_data.monitoring_user = not guild_data.monitoring_user
+
         await guild_data.save()
         get_guild_data.cache_clear()
 
-        await ctx.send(f"Monitoring is now {'enabled' if guild_data.monitoring else 'disabled'} for this guild.")
+        confirmation_message = f"Monitoring is now enabled for only {'messages' if guild_data.monitoring_message else 'users'} in this guild."
+
+        if guild_data.monitoring_message == guild_data.monitoring_user:
+            confirmation_message = f"Monitoring is now {'enabled' if guild_data.monitoring_message else 'disabled'} for both users and messages in this guild."
+        elif not guild_data.monitoring_message and not guild_data.monitoring_user:
+            confirmation_message = "Monitoring is now disabled for both users and messages in this guild."
+
+        await ctx.send(confirmation_message)
+
+    @monitor.group()
+    async def group(self, ctx: commands.Context):
+        """
+        Commands for creating, modifying, and deleting group message monitors.
+        """
+        await available_subcommands(ctx)
+
+    @group.command(name='add')
+    async def group_add(self, ctx: commands.Context, name: str, monitor_messages: ListConverter):
+        """
+        Creates new/adds to existing group message monitor.
+        """
+        if len(monitor_messages) == 0:
+            await ctx.send("You must provide at least one monitored message to add to a group.")
+            return
+
+        if monitor_messages == "*":
+            await ctx.send("You cannot add all messages to a group. Please specify individual patterns to add to the group.")
+            return
+        
+        db_monitor_messages = []
+
+        for pattern_id in monitor_messages:
+            db_monitor_message = await StaffMonitorMessage.filter(monitor_message_id=pattern_id).get_or_none()
+
+            if not db_monitor_message:
+                await ctx.send(f"The pattern with ID `{pattern_id}` could not be found.")
+                return
+
+            db_monitor_messages.append(db_monitor_message)
+        
+        monitor_group = await StaffMonitorMessageGroups.filter(name=name).get_or_none()
+
+        if not monitor_group:
+            monitor_group = await StaffMonitorMessageGroups.create(name=name)
+
+        await monitor_group.monitor_messages.add(*db_monitor_messages)
```

Comment:
Does this account for duplicates? I.e. if I try and add the same pattern twice, does anything bad happen?

Reply:
Tested for duplicates and it does not add them again to the relation table and `/monitor group list` does not show duplicates when it lists the patterns so I think it's fine. 

The only "problem" is that it still says `Messages [4, 5] have been successfully added to group test.` which is technically not correct. Then again, handling that case + the case where some values provided are already in the group and others are not might not be worth the extra code. Is this worth addressing?

Reply:
Eh nah probably not worth handling, plus I feel like it might confuse some people when they try and add a pattern to the group that already has the pattern and the bot doesn't say it added it.